tests/unit/task_system_test.c

```diff
@@ -326,6 +356,222 @@ CTEST2(task_system, test_multiple_threads)
+                           trunk_get_scratch_size(),
+                           data->tasks,
+                           data->hid,
+                           &new_thread[0]);
+   ASSERT_TRUE(SUCCESS(rc));
+
+   // Wait till 1st user-thread gets to its wait-for-stop loop
+   while (!thread_cfg[0].waitfor_stop_signal) {
+      platform_sleep(USEC_TO_NSEC(100000)); // 100 msec.
+   }
+   thread_cfg[1].tasks          = data->tasks;
+   thread_cfg[1].exp_thread_idx = task_get_max_tid(data->tasks);
+
+   // We've used up all threads. This thread creation should fail.
+   rc = task_thread_create("test_one_thread",
+                           exec_user_thread_loop_for_stop,
+                           &thread_cfg[1],
+                           trunk_get_scratch_size(),
+                           data->tasks,
+                           data->hid,
+                           &new_thread[1]);
+   ASSERT_FALSE(SUCCESS(rc),
+                "Thread should not have been created"
+                ", new_thread=%lu, max_tid=%lu\n",
+                new_thread[1],
+                task_get_max_tid(data->tasks));
+
+   // Stop the running user-thread now that our test is done.
+   thread_cfg[0].stop_thread = TRUE;
+
+   for (uint64 tctr = 0; tctr < ARRAY_SIZE(new_thread); tctr++) {
+      if (new_thread[tctr]) {
+         rc = platform_thread_join(new_thread[tctr]);
+         ASSERT_TRUE(SUCCESS(rc));
+      }
+   }
+}
+
+/*
+ * ------------------------------------------------------------------------
+ * Test that SplinterDB can be created with the task system configured with
+ * background threads.
+ * ------------------------------------------------------------------------
+ */
+CTEST2(task_system, test_splinterdb_create_w_background_threads)
+{
+   splinterdb       *kvsb;
+   splinterdb_config cfg;
+   data_config       default_data_cfg;
+
+   default_data_config_init(TEST_MAX_KEY_SIZE, &default_data_cfg);
+   create_default_cfg(&cfg, &default_data_cfg);
+
+   // Task system should be setup with background threads
+   cfg.num_normal_bg_threads   = 1;
+   cfg.num_memtable_bg_threads = 1;
+
+   int rv = splinterdb_create(&cfg, &kvsb);
+   ASSERT_EQUAL(0, rv);
+
+   splinterdb_close(&kvsb);
+}
+
+/*
+ * ------------------------------------------------------------------------
+ * Test that SplinterDB can be created even when background threads use
+ * up all the slots.
+ * ------------------------------------------------------------------------
+ */
+CTEST2(task_system, test_splinterdb_create_w_all_background_threads)
+{
+   splinterdb       *kvsb;
+   splinterdb_config cfg;
+   data_config       default_data_cfg;
+
+   default_data_config_init(TEST_MAX_KEY_SIZE, &default_data_cfg);
+   create_default_cfg(&cfg, &default_data_cfg);
+
+   // Task system should be setup with all background threads
+   cfg.num_normal_bg_threads   = (MAX_THREADS - 2);
+   cfg.num_memtable_bg_threads = 1;
+
+   int rv = splinterdb_create(&cfg, &kvsb);
```

Comment:
Same comment as above.  These tests only need the public API.  Having them here creates confusion about which units should be covered by which unit tests.

Reply:
Relocated these two test cases to `splinterdb_quick_test.c`

Reply:
Also pulled in the suggested code in next comment to start of CTEST_SETUP() as suggested.